core/src/main/scala/org/apache/spark/status/KVUtils.scala

```diff
@@ -111,7 +122,7 @@ private[spark] object KVUtils extends Logging {
         // The default serializer is slow since it is using JSON+GZip encoding.
         Some(new KVStoreProtobufSerializer())
       } else {
-        None
+        Some(serializerForHistoryServer(conf))
       }
```

Comment:
QQ: If it is always non empty, change from Option ?

Reply:
Thanks, I just did some refactoring. Please take another look